contributors/design-proposals/auth/pod-restriction.md

```diff
@@ -0,0 +1,477 @@
+
+## Scope
+
+This proposal is about building a new policy type that can place restrictions on any or all pod
+fields (see [Monolith vs. Modular](#monolith-vs-modular)). However, the goal is not to combine all
+admission controllers into a single one. The following controls are explicitly out of scope:
+
+-   Defaulting (and mutations in general) - See [No Defaulting](#no-defaulting).
+
+-   External dependencies - The policy doesn't express any inter-object relationships, and doesn't
+    call out to external servers or have plugin points.
+
+-   Complex logic or interfield relationships - The goal is to keep the policy simple, both to
+    implement and reason about. The only exception is fields that explicitly reference others like
+    VolumeMounts and EnvVarSource, but most fields should be able to be validated without knowledge
+    of any other Pod fields.
+
+## Design
+
+### Policy Binding
+
+PodRestrictions are applied at the namespace level. Any Pod (or PodTemplate object) must conform to
+the policies matching the namespace it is created in.
+
+```go
+Binding:
+    Namespaces        []string
+    NamespaceSelector *metav1.LabelSelector
+```
+
+#### Justification
+
+There are a few different approaches taken by our current policies (see
+[#60001](https://github.com/kubernetes/kubernetes/issues/60001)).
+
+**Bind to Users**: Binding policy to users is the obvious first choice, but has a number of
+problems. First, users don't typically create pods, they create resources like Deployments that
+create pods on their behalf. That "on behalf" ends up being problematic to implement, especially
+with issues like confused deputy attacks, third party controllers, and changing policies.
+
+**Bind to ServiceAccounts**: To get around the user problem, PodSecurityPolicy adds the ability to
+bind to the target objects service account. ServiceAccount usage isn't actually authorized
+separately from namespace usage, so this just ends up being namespace-level authorization. However,
+because PodSecurityPolicy is authorized by either the source or target user, this ends up being less
+secure, since the Pod can use it's service account to create pods in other namespaces with the same
+access level. For instance, using PodSecurityPolicy you cannot have a controller that runs as
+privileged without opening up all pods it creates to running as privileged.
+
+**Namespaced Objects**: This approach has its advantages, but it doesn't scale well to large
+organizations with many namespaces (without a system on top managing all the policies across
+namespaces). See extensions below for how we might use namespaced objects with PodRestrictions.
+
+**Bind to Pod labels**: Pod labels don't have any authorization of usage without something like
+PodRestriction, so binding to labels isn't sufficient no it's own.
+
+**Bind to Namespaces**: The approach I'm recommending. Cluster level objects use a namespace
```

Comment:
This is much more intuitive. I am not sure how well it works at scale. One case I could think of is having a lot of namespaces all having different labels based on which selection could happen resulting in delay of policy binding. This could be avoided by labeling all the namespaces with predefined value though.